src/components/engagement/handlers/update-project-status.handler.ts

```diff
@@ -13,35 +14,35 @@ export class UpdateProjectStatusHandler
     private readonly engagementService: EngagementService
   ) {}
 
-  async handle({ previous, updated, updates, session }: ProjectUpdatedEvent) {
-    // When the project becomes Active if it is currently InDevelopment
-    if (
-      previous.status !== ProjectStatus.InDevelopment ||
-      updated.status !== ProjectStatus.Active
-    ) {
-      return;
-    }
-
-    // Update engagement status to Active
-    const engagements = await this.db
+  private async getEngagements(projectId: string) {
+    return await this.db
       .query()
-      .match([node('project', 'Project', { id: updates.id })])
+      .match([node('project', 'Project', { id: projectId })])
       .match([
         node('project'),
         relation('out', '', 'engagement', { active: true }),
         node('engagement'),
+        relation('out', '', 'status', { active: true }),
+        node('sn', 'Property'),
       ])
-      .return('engagement.id as id')
-      .asResult<{ id: string }>()
+      .return('engagement.id as id, sn.value as status')
+      .asResult<{ id: string; status: string }>()
```

Comment:
```suggestion
      .asResult<{ id: string; status: EngagementStatus }>()
```
Then you can drop the typecast below